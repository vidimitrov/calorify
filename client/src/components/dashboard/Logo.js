import styled, { keyframes } from 'styled-components';

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Logo = styled.img.attrs({
  alt: 'logo',
})`
  animation: ${rotate} infinite 20s linear;
  height: 40vmin;
  src: url(${props => props.src});
`;

export default Logo;
