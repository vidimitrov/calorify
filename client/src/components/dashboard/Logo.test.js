import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Logo from './Logo';

it('Dashboard/Logo component renders correctly', () => {
  const tree = renderer.create(<Logo src="../../assets/img/logo.png" />).toJSON();
  expect(tree).toMatchSnapshot();
});
