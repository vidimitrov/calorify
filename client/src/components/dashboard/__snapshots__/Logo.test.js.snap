// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard/Logo component renders correctly 1`] = `
.c0 {
  height: 25px;
  margin-right: 10px;
  src: url(../../assets/img/logo.png);
}

<img
  alt="logo"
  className="c0"
  src="../../assets/img/logo.png"
/>
`;
